packages/backend-storage/src/construct.ts

```diff
@@ -153,20 +155,26 @@ export class AmplifyStorage
    * Store storage outputs using provided strategy
    */
   private storeOutput = (
-    outputStorageStrategy: BackendOutputStorageStrategy<StorageOutput> = new StackMetadataBackendOutputStorageStrategy(
-      Stack.of(this)
-    ),
+    outputStorageStrategy: BackendOutputStorageStrategy<
+      StorageOutput | StorageBucketsPayload
```

Comment:
we shouldn't create `StorageBucketsPayload`. I.e. we shouldn't introduce storage specific view of its output schema. 

Instead

We should add type like this one https://www.geodev.me/blog/deeppartial-in-typescript to `plugin-types` and then make `appendToBackendOutputList` take `DeepPartial<BackendOutput>` . I.e. loosen abstraction.

Reply:
I was not sure if to add the new type (or `DeepPartial<BackendOutput>`) to storage schema or to `plugin-types` because `StorageOutput` is in storage schema. How do we decide where to add?

Reply:
This API should be modified to accept partials https://github.com/aws-amplify/amplify-backend/blob/ec12ac83d730a37c539ff6ae2ee282341d4c5342/packages/plugin-types/src/output_storage_strategy.ts#L8 .
Hence type should be defined to be accessible to it.

Reply:
Done [88ea17d](https://github.com/aws-amplify/amplify-backend/pull/1742/commits/88ea17da1f11c5eccfa655be9d00ec87515df95f) Thanks for help!